labs/lab08/sol/lin/6-once/once.c

```diff
@@ -0,0 +1,162 @@
+/**
+ * SO
+ * Lab #8
+ *
+ * Task #6, lin
+ *
+ * Implementing single time use init / deinit functions
+ */
+#include <pthread.h>
+#include <unistd.h>
+#include "utils.h"
+
+#define ONCE_INITIALIZER { PTHREAD_MUTEX_INITIALIZER, 0 }
+
+#define NUM_THREADS 10
+#define NUM_CHARS   1000
+#define START_CHAR  '!'
+#define PERIOD      500
+
+struct once_struct {            /* Our equivalent of pthread_once_t */
+	pthread_mutex_t mutex;
+	int refcount;
+};
+
+struct once_struct once = ONCE_INITIALIZER;
+
+/**
+ * Remaining code is for testing one_time_init()
+ * Do NOT edit this function
+ */
+static void init_func(void)
+{
+	/*
+	 * We should see this message only once, no matter how many
+	 * times one_time_init() is called
+	 */
+
+	time_t ts = time(NULL);
+
+	printf("Threads execution started at %s", asctime(gmtime(&ts)));
+}
+
+static void deinit_func(void)
+{
+
+	time_t ts = time(NULL);
+
+	/*
+	 * We should see this message only once, no matter how many
+	 * times one_time_init() is called
+	 */
+
+	printf("\nThreads execution ended at %s", asctime(gmtime(&ts)));
+}
+
+static void thread_work(int arg)
+{
+    for (int i = 0; i < NUM_CHARS; i++) {
+        printf("%c", START_CHAR + arg);
+
+        /* What happens if you comment out the fflush ? */
+
+        fflush(stdout);
+
+        /* What happens if you decrease the PERIOD ? */
+
+        usleep(PERIOD);
+    }
```

Comment:
```suggestion
	for (int i = 0; i < NUM_CHARS; i++) {
		printf("%c", START_CHAR + arg);

		/* What happens if you comment out the fflush? */

		fflush(stdout);

		/* What happens if you decrease the PERIOD? */

		usleep(PERIOD);
	}
```

Apply this to `lab08/skel/lin/6-once/once.c` as well.